File: (7 kyu) Mumbling/(7 kyu) Mumbling.ts
export class G964 {
  public static accum(s: string): string {
    return s
      .split('')
      .map((el, i) =>
        /^[a-zA-Z]$/ ? el.toUpperCase() + el.repeat(i).toLowerCase() : '',
      )
      .join('-');
  }
}

File: (7 kyu) Mumbling/(7 kyu) Mumbling.go
package kata

import "strings"

func Accum(s string) string {
  var sb []string
  letters := strings.Split(s, "")

  for i, letter := range letters {
    s := strings.Title(strings.Repeat(strings.ToLower(letter), i+1))
    sb = append(sb, s)
  }

  return strings.Join(sb, "-")
}
